README.md

```diff
@@ -182,6 +182,102 @@ You can intermittently poll whether the operation has finished by using the `Pol
   }
 ```
 
+#### `Azure::Core::Context`
+
+Most Azure SDK Service Client methods accept an optional `Azure::Core::Context` parameter, which is used to enable cancellation of the operation or to
+establish an absolute deadline for the operation.
+
+This is useful when you want to assign a time limit on an operation to ensure that it completes in a "reasonable" timeframe. For instance, the 
+snippet below will cancel a blob client upload after 5 seconds.
+
+```cpp
+  Azure::Core::Context cancelledIn5s = Azure::Core::Context::ApplicationContext.WithDeadline(
+          std::chrono::system_clock::now() + std::chrono::seconds(5);
+
+  std::cout << "Uploading blob: " << blobName << std::endl;
+  try {
+    blobClient.UploadFrom(blobContent, sizeof(blobContent), cancelledIn5s);
+  }
+  catch (const Azure::Core::OperationCancelledException& e) {
+	std::cout << "Upload was cancelled: " << e.what() << std::endl;
+  }
+```
+
+`Context` objects can also be directly cancelled using the `Cancel()` method.
+
+`Context` objects form a directed tree, where a child context can be created from a parent context. 
+The context tree is unidirectional and acyclic.
+
+There are five basic operations that can be performed on a 'Context' object:
+
+* Create a child context from a parent context which can be independently cancelled.
+* Create a child context from a parent context with a Key/Value pair. This is useful for associating metadata with a context.
+* Create a child context from a parent context with a deadline. This is useful for setting a timeout.
+* Cancel a context. This will cancel the context and all its children. Note that there is no way of un-cancelling a context.
+* Check if a context is cancelled.
+
+By default, when a `Context` is created, it is a child of the `ApplicationContext`. This allows a client application to cancel all the operations currently active in the process.
+
+If you do NOT want a newly created `Context` to be a child of `ApplicationContext`, use the `Azure::Core::Context::CreateNewRoot()` function, 
+which will create a new root context.
+
+When a context is copied from another context, the copied context will share state with the original context. 
+This means that if the original context is cancelled, the copied context will also be cancelled.
+
+Cancellation of a `Context` is a permanent operation. Once a non-root context is cancelled, it cannot be un-cancelled.
+
+Root `Contexts` are unique in that their cancellation *CAN* be reset. Cancelling a root context essentially creates a new root Context replacing the existing root.
```

Comment:
The semantic distinction between Root and non-root contexts is implicit in the design for Context. Whenever you call "WithDeadline", "WithValue", or "WithContext", you are explicitly creating a parent/child relationship between contexts. And that defines a tree. 

And every tree needs a root.

There is no such thing as a "Regular" context.

All Context object have a parent object. The core aspect of this change is that in previous versions, the relationship between a newly created Context object and ApplicationContext was described in the documentation but not in the implementation. 

This change makes the relationship described in the documentation explicit.

And there is basically no situation where an end-user would want to go down that path. They simply don't need to know these details.

What an end-user needs to know is:
1) Every API takes a Context object.
1) If you want to cancel a call, cancel the Context for the call.
1) If you want to cancel ALL calls in the process, cancel ApplicationContext.
    1) If you have cancelled ApplicationContext, you can call ApplicationContext::Reset to start new operations, otherwise your application is toast.

The `CreateNewRoot` method exists for two reasons:

1) It allows us to create ApplicationContext, and
2) It allows a customer who wants the old behavior back to implement the functionality from before the change.